designs/extensible-cli-and-scaffolding-plugins-phase-1.md

```diff
@@ -135,6 +113,36 @@ type Deprecated interface {
 }
 ```
 
+## Configuration
+
+### Config version `3-alpha`
+
+Any changes that break `PROJECT` file backwards-compatibility require a version
+bump. This new version will be `3-alpha`, which will eventually be bumped to
+`3` once the below config changes have stabilized.
+
+### Project file plugin `layout`
+
+The `PROJECT` file will specify what base plugin generated the project under
+a `layout` key. `layout` will have the format: `Plugin.Name() + "/" + Plugin.Version()`.
+`version` and `layout` have versions with different meanings: `version` is the
+project config version, while `layout`'s version is the plugin semantic version.
+The value in `version` will determine that in `layout` by a plugin's supported
+project versions (via `SupportedProjectVersions()`).
+
+Example `PROJECT` file:
+
+```yaml
+version: "3-alpha"
```

Comment:
1. Will not the projects built with the current kubebuilder implementation still working after the plugins?
2. If not, what are the steps required for the users to upgrade the current projects to the new version? 

Also, I understand that:
The V1 to V2 -> means break changes in the scaffold of the projects. 
So, what is the breaking change which justifies the V3? 

Should we not just bump the new tag 3.0.0 in order to make clear the breaking changes in the commands of the cli tool? And, should we really use 3-alpha? Had the kubebuilder the 2-alpha as well? 

c/c @mengqiy @DirectXMan12

Reply:
Based in the comment https://github.com/kubernetes-sigs/kubebuilder/pull/1403/files#r404405621 I am assuming that everybody is ok with . 👍